bundles/org.openhab.binding.ws980wifi/src/main/java/org/openhab/binding/ws980wifi/internal/ws980wifiHandler.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ws980wifi.internal;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ws980wifi.internal.discovery.ws980wifi;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tech.units.indriya.unit.Units;
+
+/**
+ * The {@link ws980wifiHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joerg Dokupil - Initial contribution
+ */
+@NonNullByDefault
+public class ws980wifiHandler extends BaseThingHandler {
+
+    private final Logger log = LoggerFactory.getLogger(ws980wifiHandler.class);
+    private @Nullable ScheduledFuture<?> pollingJob = null;
+    private @Nullable ws980wifiConfiguration config;
+    private String host = "";
+    private String port = "";
+
+    public ws980wifiHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // sensor things cannot send any messages, hence they are not allowed to handle any command
+        // The only possible command would be "Refresh"
+    }
+
+    @Override
+    public void initialize() {
+        Thing thing = getThing();
+        if (thing == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Thing is NULL");
+        } else {
+            config = getConfigAs(ws980wifiConfiguration.class);
+            host = config.getHost();
+            port = config.getPort();
+
+            pollingJob = scheduler.scheduleWithFixedDelay(this::updateWeatherData, 0, config.refreshInterval,
+                    TimeUnit.SECONDS);
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
Too early to conclude that it's online. If you set ThingStatus.UNKNOWN before scheduling the job, you can set ThingStatus.ONLINE in updateWeatherData after successfully communicating with the device.